### ISSUE TITLE
Invalid Java Gradle Compatibility Link in Flutter CLI Fix

### ISSUE THREAD

**Issue by EchoEllet** [flutter/flutter#156282]
> [!NOTE]
> **This is not a bug or issue in Flutter**, however, the link is probably outdated as opening it will result in 404.

**Correct link**: https://docs.flutter.dev/to/java-gradle-incompatibility
**Invalid link (suggested by Flutter CLI)**: https://flutter.dev/to/to/java-gradle-incompatibility

1. Update **Android Studio** to the latest version that uses Java 21 as a minimum, Update Flutter to the latest version
2. Use an old Flutter project that uses Gradle version `7.6.3` or create a new one, navigate to the `android` directory, and run `./gradlew wrapper --gradle-version=7.6.3`. You could also use any Gradle version that doesn't support Java 21.
3. Run the Android app:

```console
┌─ Flutter Fix ────────────────────────────────────────────────────────────────────────────────────┐
│ [!] Your project's Gradle version is incompatible with the Java version that Flutter is using    │
│ for Gradle.                                                                                      │
│                                                                                                  │
│ If you recently upgraded Android Studio, consult the migration guide at                          │
│ https://flutter.dev/to/to/java-gradle-incompatibility.                                           │
│                                                                                                  │
│ Otherwise, to fix this issue, first, check the Java version used by Flutter by running `flutter  │
│ doctor --verbose`.                                                                               │
│                                                                                                  │
│ Then, update the Gradle version specified in                                                     │
│ ~/Developer/playground/framework_based/flutter/flutter-quill/example/android/gradle/w │
│ rapper/gradle-wrapper.properties to be compatible with that Java version. See the link below for │
│ more information on compatible Java/Gradle versions:                                             │
│ https://docs.gradle.org/current/userguide/compatibility.html#java                                │
2
│                                                                                                  │
└──────────────────────────────────────────────────────────────────────────────────────────────────┘
```
4. Open the link https://flutter.dev/to/to/java-gradle-incompatibility and you will get 404.

<details><summary>Flutter Doctor result</summary>
<p>

```console
[✓] Flutter (Channel stable, 3.24.3, on macOS 14.6.1 23G93 darwin-arm64, locale en-US)
    • Flutter version 3.24.3 on channel stable at ~/sdks/flutter
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision 2663184aa7 (3 weeks ago), 2024-09-11 16:27:48 -0500
    • Engine revision 36335019a8
    • Dart version 3.5.3
    • DevTools version 2.37.3

[✓] Android toolchain - develop for Android devices (Android SDK version 34.0.0)
    • Android SDK at ~/Library/Android/sdk
    • Platform android-34, build-tools 34.0.0
    • Java binary at: ~/Applications/Android Studio.app/Contents/jbr/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 21.0.3+-79915917-b509.11)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 16.0)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Build 16A242d
    • CocoaPods version 1.15.2

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 2024.2)
    • Android Studio at ~/Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 21.0.3+-79915917-b509.11)

[✓] IntelliJ IDEA Community Edition (version 2024.2.3)
    • IntelliJ at ~/Applications/IntelliJ IDEA Community Edition.app
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart

[✓] VS Code (version 1.93.1)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.98.0

[✓] Connected device (4 available)
    • sdk gphone64 arm64 (mobile)     • emulator-5554         • android-arm64  • Android 14 (API 34) (emulator)
    • macOS (desktop)                 • macos                 • darwin-arm64   • macOS 14.6.1 23G93 darwin-arm64
    • Mac Designed for iPad (desktop) • mac-designed-for-ipad • darwin         • macOS 14.6.1 23G93 darwin-arm64
    • Chrome (web)                    • chrome                • web-javascript • Google Chrome 129.0.6668.90

[✓] Network resources
    • All expected network resources are available.

• No issues found!
```

</p>
</details> 

<details><summary>Screenshot</summary>
<p>

![image](https://github.com/user-attachments/assets/0a8f7b6d-45a3-4074-82e3-941b12a3a87e)

</p>
</details>

**Comment by stuartmorgan** [comment]
This has already been fixed on `master`, via https://github.com/flutter/flutter/pull/150999, so closing as resolved.

**Comment by thookuchatti** [comment]
So how to solve this issue? i cant able to run my application.

**Comment by EchoEllet** [comment]
I just fixed it in [Flutter Quill #2312](https://github.com/singerdmx/flutter-quill/pull/2312) (see the changes).

The issue is that the latest version of Android Studio (Ladybug 2024.2.1) bundles Java 21 which is supported on Gradle 8.5 and newer versions, using newer versions of Gradle requires also updating the Android Gradle Plugin (AGP), also some minor changes to the build configuration, updating Kotlin version to `1.8.22` (the new Flutter default) was required for my case, Java 21 deprecate the support for Java 8 and if you use any Flutter plugins that still targets Java 8 you will get warnings.
